### ISSUE TITLE
Error when working with strict.lua

### ISSUE THREAD

**Issue by shmuz** [Stepets/utf8.lua#6]
```
stepets\utf8\functions\lua53.lua:37: variable 'plain' is not declared
```
This is a minor issue, more serious ones will hopefully follow.
Each next bug is getting harder to find.

**Comment by Stepets** [comment]
Took strict.lua from here http://metalua.luaforge.net/src/lib/strict.lua.html
Fixed all what it found

**Comment by shmuz** [comment]
There is a serious issue probably related  to caching. A matching functions in the complex system can fail to match while the same calls in a simple standalone script succeed.

How can I turn off the caching?

**Comment by Stepets** [comment]
Seems like I forgot to re-enable caching, so it's turned off already.
It might be caused by coroutines, but I'm pretty sure there is no global state in library.
Can you provide setup that not working as expected? Maybe I can help to localize bug.

**Comment by Stepets** [comment]
You can write me directly. My telegram username: `@stepets`

**Comment by shmuz** [comment]
Sorry I've no telegram. BTW, if you prefer to communicate in Russian it is possible with me.
Do you have a Windows machine?
If yes I can provide a setup (that includes Far Manager).

**Comment by Stepets** [comment]
Yes, I have windows machine.

**Comment by shmuz** [comment]
Ok then let's resume tomorrow. Good night.

**Comment by shmuz** [comment]
1. Download Far Manager (further: "Far") as a 7z-archive from https://farmanager.com/download.php?l=en and unpack it into directory `Far3` that you created in advance
2. Rename `Far3\Far.exe.example.ini` to `Far.exe.ini`
3. In this file uncomment the line `;UseSystemProfiles=1` (remove ;) and change 1 to 0
4. Create directory `Far3\Lua\stepets\utf8` and place your library there
5. Run `Far.exe`. It will create directory `Far3\Profile` with subdirectories.
6. Create file `Far3\Profile\luafar_init.lua` with the following contents:
```Lua
local u = (require "stepets.utf8"):init()
for k,v in pairs(u) do utf8[k]=v end
```
7. Create file `Far3\test.lua` containing `far.Message "Hello"`
8. Restart Far (exit it via `F10` then run again).
9. When you are in `Far3` directory run from the Far command line `lua:@test.lua`. The red error message box should appear which says `Invalid macrosequence specification`.

When all this works we will continue. If there are problems please let me know.

**Comment by Stepets** [comment]
Got red error message box as you described. What's next?

**Comment by shmuz** [comment]
This error message box is due to pattern matching library replacement. It does not occur normally given the valid input `lua:@test.lua` which should run the file `test.lua` in the active panel.

See function `GetFileParams` in file `Far3\Plugins\LuaMacro\luamacro.lua`.

For debugging you can use the function `far.Show` which accepts arbitrary number of arguments of any Lua type and shows their values on the screen.

After any change in `luamacro.lua` you should either restart Far or run the following command from the Far command line: `lua:_G.RecreateLuaState=true`

You can use `Alt-F8` for selecting previously run commands.

Ask me for additional help if needed.

**Comment by shmuz** [comment]
A new info that was found today (it may be helpful):
If we replace `Far3\lua51.dll` (which is LuaJIT 2.1) with a standard lua51.dll then `lua:@test.lua` executes correctly.

**Comment by Stepets** [comment]
Fixed this somehow in 6731f8e8854f3b39fc20bbbf4216352440a731f7

**Comment by shmuz** [comment]
- Now it passes all my Far Manager tests provided LuaJIT is built by Visual Studio
- Fails with MinGW LuaJIT build
- Fails with standard lua51.dll (no FFI)

I hope my reports were helpful.
